### PR TITLE
docs: Add PyCharm documentation

### LINKED ISSUE #189: PyCharm integration
Here's a Pixi variant of my https://github.com/jonashaag/micromamba-pycharm.

Put this into each `.pixi/` folder as `.pixi/conda`.

```py
#!/usr/bin/env python3

import sys
import json
import subprocess
import yaml
from pathlib import Path

PIXI_ROOT = Path(__file__).parent
PROJECT_ROOT = PIXI_ROOT.parent

ENABLE_LOGGING = True
if ENABLE_LOGGING:
    open("/tmp/conda-wrapper-log", "a").write(str(sys.argv[1:]) + "\n")

if sys.argv[1:4] == ["info", "--envs", "--json"]:
    print(
        json.dumps(
            {
                "envs_dirs": [str(PIXI_ROOT)],
                "conda_prefix": str(PIXI_ROOT),
                "envs": [str(PIXI_ROOT / "env")],
            }
        )
    )
elif sys.argv[1] == "list":
    lockfile = yaml.safe_load((PROJECT_ROOT / "pixi.lock").read_text())
    for pkg in sorted(lockfile["package"], key=lambda pkg: pkg["name"]):
        print(pkg["name"], pkg["version"], pkg["build"], "fake-channel")
elif sys.argv[1] == "run":
    assert sys.argv[4] == "--no-capture-output"
    subprocess.run(["pixi", "run", *sys.argv[5:]])
else:
    raise NotImplementedError(str(sys.argv[1:]))
```

### PR DESCRIPTION
**Pull request by pavelzw**
Closes #189 

https://github.com/pavelzw/pixi-pycharm

### CODE REVIEW

File: docs/ide_integration/pycharm.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    If you get the error `could not determine any available versions for pixi-pycharm on win-64` when running `pixi add pixi-pycharm` (even when you're not on Windows),
+    this is because the package is not available on Windows and pixi tries to solve the environment for all platforms.
+    If you still want to use it in your pixi project (and are on Linux/macOS), you can add the following to your `pixi.toml`:
+
+    ```toml
+    [target.osx-arm64.dependencies] # (1)!
+    pixi-pycharm = "*"
+    ```
+
+    1. Or `[target.linux-64.dependencies]` depending on your platform.
```
Comment (ruben-arts) [inline docs/ide_integration/pycharm.md:36]: You can also do `target.unix`

File: docs/ide_integration/pycharm.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+This is an executable that tricks PyCharm into thinking it's the proper `conda` executable.
+Under the hood it redirects all calls to the corresponding `pixi` equivalent.
+
+!!!warning "Use the conda shim from this pixi project"
+    Please make sure that this is the `conda` shim from this pixi project and not another one.
+    If you use multiple pixi projects, you might have to adjust the path accordingly as PyCharm remembers the path to the conda executable.
+
+![Add Python Interpreter](https://raw.githubusercontent.com/pavelzw/pixi-pycharm/main/.github/assets/add-conda-environment-light.png#only-light)
+![Add Python Interpreter](https://raw.githubusercontent.com/pavelzw/pixi-pycharm/main/.github/assets/add-conda-environment-dark.png#only-dark)
```
Comment (ruben-arts) [inline docs/ide_integration/pycharm.md:57]: Nice going the extra mile for light mode support

**Review by ruben-arts — COMMENTED** [review]
Awesome 😎

File: docs/ide_integration/pycharm.md (hunk 1 above)
Comment (pavelzw) [reply docs/ide_integration/pycharm.md:36]: Good to know :)

**Comment by ruben-arts** [comment]
Just installed PyCharm to test it, and it works "like a charm" :stuck_out_tongue_closed_eyes: (sorry)

Kudos!

**Pull request merged**